system-images/compat/src/config/mod.rs

```diff
@@ -1,26 +1,52 @@
 use std::borrow::Cow;
-use std::collections::BTreeMap;
+use std::collections::{BTreeMap, BTreeSet, HashMap};
 use std::path::Path;
 
 use beau_collector::BeauCollector;
 use embassy::config::action::SetResult;
 use embassy::config::{spec, Config};
+use embassy::s9pk::manifest::PackageId;
+use embassy::status::health_check::HealthCheckId;
 use linear_map::LinearMap;
 
 pub mod rules;
 
 use anyhow::anyhow;
 pub use rules::{ConfigRuleEntry, ConfigRuleEntryWithSuggestions};
 
+use self::rules::ConfigRule;
+pub type DepInfo = HashMap<PackageId, DepRuleInfo>;
+#[derive(Clone, Debug, serde::Deserialize)]
+pub struct DepRuleInfo {
+    condition: ConfigRule,
+    health_checks: BTreeSet<HealthCheckId>,
+}
+
 pub fn validate_configuration(
     name: &str,
     config: Config,
     rules_path: &Path,
     config_path: &Path,
+    maybe_deps_path: Option<&str>,
 ) -> Result<SetResult, anyhow::Error> {
     let rules: Vec<ConfigRuleEntry> = serde_yaml::from_reader(std::fs::File::open(rules_path)?)?;
     let mut cfgs = LinearMap::new();
     cfgs.insert(name, Cow::Borrowed(&config));
+
+    let mut depends_on = BTreeMap::new();
+    if maybe_deps_path.is_some() {
+        let deps_path = Path::new(maybe_deps_path.unwrap());
```

Comment:
```suggestion
    if let Some(deps_path) = maybe_deps_path.map(Path::new) {
```

Reply:
I can't grok this pattern for the life of me 😣 will update

Reply:
matches also work if you forget the `if let` construct.

Reply:
also:

```rust
if let Pat(...) = expr {
    do_x();
} else {
    do_y();
}
```
is *exactly* identical to
```rust
match expr {
    Pat(...) => { do_x(); }
    _ => { do_y(); }
}
```